ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Query.hs

```diff
@@ -162,6 +164,13 @@ getIsInvalidBlock
 getIsInvalidBlock CDB{..} =
   fmap (fmap (fmap invalidBlockReason) . flip Map.lookup) <$> readTVar cdbInvalid
 
+getIsValidated
+  :: forall m blk. (IOLike m, HasHeader blk)
+  => ChainDbEnv m blk
+  -> STM m (RealPoint blk -> Bool)
+getIsValidated CDB{..} =
+    flip Set.member <$> LgrDB.getPrevApplied cdbLgrDB
+
```

Comment:
To test it: pick a random block (in most cases in the ChainDB, in some cases not, use labelling), then verify the three outcomes:

* `Just x`: check that it agrees with the model that it is valid or not.
* `Nothing`: verify that it's not in the current fragment.